peer/brontide.go

```diff
@@ -1661,6 +1661,26 @@ func (p *Brontide) handleCustomMessage(msg *lnwire.Custom) error {
 	return p.cfg.HandleCustomMessage(p.PubKey(), msg)
 }
 
+// isLoadedFromDisk returns true if the provided channel ID is loaded from
+// disk.
+func (p *Brontide) isLoadedFromDisk(chanID lnwire.ChannelID) bool {
+	// If this is a newly added channel, no need to reestablish.
+	_, added := p.addedChannels.Load(chanID)
+	if added {
+		return false
+	}
+
+	// Return false if the channel is unknown.
+	channel, ok := p.activeChannels.Load(chanID)
+	if !ok {
+		return false
+	}
+
+	// During startup, we will use a nil value to mark a pending channel
+	// that's loaded from disk.
+	return channel == nil
```

Comment:
Ah yeah, this is already in place (nil values in the map) and is indeed the first location we started to use that trick...

Reply:
Added a new map to handle this specific case.

Reply:
Tried to implement here, a commit [here](https://github.com/yyforyongyu/lnd/commit/4f033c6a4f4abf2eb6031d0b99e96f4595b18c9a) in a different branch, 

To remove the nil usage, we need three maps,
- `activeChannels` to store active channels only
- `pendingChannels` to store pending channels only
- `reestablishChannels` to store pending channels loaded from disk

I think that's quite some maps...meanwhile I noticed the channel store already have method `lnChan.IsPending()` to decide whether it's pending or not. This leads me to think we should manage the channel state inside funding manager only, and access it from brontide, instead of maintaining yet another state of the channel.

This means we should expand `lnChan` to have methods so the brontide can know its state, like whether it's marked open or channel ready sent. Then inside brontide, we can decide what to do based on the state of the channel saved in `activeChannels`, wdyt?

Reply:
> Can defer convo to a refactor PR, but I don't like using lnChan because it's in-memory.

Agree here. I think we wanna slowly move the subsystems to deal with the db directly, treating the disk records as the single source of truth. Atm we can already do this in `brontide` as it has access to the db instance, maybe in another PR.